app/src/main/java/cm/aptoide/pt/home/bundles/BundlesAdapter.java

```diff
@@ -296,4 +297,21 @@ public boolean isLoaded() {
     }
     return true;
   }
+
+  public void updateAppComingSoonCard(AppComingSoonPromotionalBundle homeBundle) {
+    for (HomeBundle bundle : bundles) {
+      if (bundle.getType()
+          .equals(HomeBundle.BundleType.APP_COMING_SOON)) {
+        if (((AppComingSoonPromotionalBundle) bundle).getActionItem()
+            .getUrl()
+            .equals(homeBundle.getActionItem()
+                .getUrl())) {
+          int index = bundles.indexOf(bundle);
+          ((AppComingSoonPromotionalBundle) bundle).setRegisteredForNotification(
+              homeBundle.isRegisteredForNotification());
+          notifyItemChanged(index);
+        }
+      }
+    }
```

Comment:
I may be oversimplifying but... can't this be replaced by?:

```suggestion
    int index = bundles.indexOf(homeBundle);
    ((AppComingSoonPromotionalBundle) bundles.get(index)).setRegisteredForNotification(homeBundle.isRegisteredForNotification());
    notifyItemChanged(index);
```